devito/finite_differences/differentiable.py

```diff
@@ -115,6 +120,14 @@ def _eval_at(self, func):
             return self
         return self.func(*[getattr(a, '_eval_at', lambda x: a)(func) for a in self.args])
 
+    @property
+    def _eval_deriv(self):
+        return self.func(*[getattr(a, '_eval_deriv', a) for a in self.args])
+
+    @property
+    def _fd_priority(self):
+        return .75 if self.is_TimeDependent else .5
```

Comment:
magic numbers?

Reply:
how about, instead of this property, we just add it to `TimeFunction` and `Function` respectively?

Reply:
They have one, this is an Expression and needs its own priority that is lower than the Function one.

Reply:
ah right sorry I'm dumb